### PR TITLE
DSD-635: GA Linter update

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-635](https://jira.nypl.org/browse/DSD-635)
## This PR does the following:
- Fixes the Github Actions "CI / Lint" task.
- Pins the Typescript version. The previous unpinned version caused incompatibility with an eslint plugin and caused a warning to appear and the lint task to fail.
- Fixes minor linting issues in a few files.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr791-y5oveuer6fol625skmgtdziowvsphepj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61aa8462aee91c8f50c53ede

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr791-y5oveuer6fol625skmgtdziowvsphepj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61aa8462aee91c8f50c53ede

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ Currently, this repo is in Prerelease. When it is released, this project will ad
 - Exports components and variables that were initially missed when they were added: `ColorVariants`, `Fieldset`, `IconAlign`, `StatusBadgeTypes`,
 - Fixes `SearchBar` by passing necessary props down to its `TextInput` through the `textInputProps` prop.
 - Fixes `DatePicker` component unit tests.
+- Fixes the Github Actions "CI / Lint" task by pinning the Typescript version.
```
Comment (bigfishdesign13) [inline CHANGELOG.md]: What not wrap `CI / Lint` in backticks?

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.  One minor question.

File: CHANGELOG.md (hunk 1 above)
Comment (EdwinGuzman) [reply CHANGELOG.md]: I can, I typically leave it for component or variable names.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr791-y5oveuer6fol625skmgtdziowvsphepj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61aa8462aee91c8f50c53ede

**Pull request merged**